presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedSet.java

```diff
@@ -130,9 +135,29 @@ public void testBigintSimpleTypedSet()
         }
     }
 
+    @Test
+    public void testMemoryExceeded()
+            throws Exception
+    {
+        try {
+            TypedSet typedSet = new TypedSet(BIGINT, 10, FUNCTION_NAME);
+
+            for (int i = 0; i <= TypedSet.FOUR_MEGABYTES + 1; i++) {
+                Block block = createLongsBlock(Collections.nCopies((int) 1, (long) i));
+
+                typedSet.add(block, 0);
+            }
+
+            fail("expected exception");
+        }
+        catch (PrestoException e) {
```

Comment:
i went this route because i'm not aware of how to used expectedExceptions to verify fields in an exception